cupy/core/dlpack.pyx

```diff
@@ -234,10 +257,14 @@ cpdef ndarray fromDlpack(object dltensor) except +:
     shape_vec.assign(shape, shape + ndim)
 
     if mem.dlm_tensor.dl_tensor.strides is NULL:
+        # Make sure this capsule will never be used again.
+        cpython.PyCapsule_SetName(mem.dltensor, 'used_dltensor')
```

Comment:
The name change is deferred to right before `return` as errors could be raised anywhere before arriving here (or below), and technically the tensor is not consumed.